tensorflow/stream_executor/cuda/cuda_dnn.cc

```diff
@@ -334,15 +342,18 @@ port::Status CudnnSupport::Init() {
     // Check whether loaded version of CuDNN matches what the source
     // was built with.
     size_t loaded_version = dynload::cudnnGetVersion();
-    bool library_loaded_matches_source = (loaded_version == CUDNN_VERSION);
+    size_t loaded_major_version = cudnnMajorVersion(loaded_version);
+    bool library_loaded_matches_source =
+        (loaded_major_version == cudnnMajorVersion(CUDNN_VERSION));
```

Comment:
I would do "version / 100 * 100" and call it "compatibility version". That
way, it is still similar to the original version in terms of scale.

On Tue, Jul 12, 2016 at 10:43 PM, Vijay Vasudevan <notifications@github.com>
wrote:

> In tensorflow/stream_executor/cuda/cuda_dnn.cc
> <https://github.com/tensorflow/tensorflow/pull/3249#discussion_r70566758>:
>
> > @@ -334,15 +342,18 @@ port::Status CudnnSupport::Init() {
> >      // Check whether loaded version of CuDNN matches what the source
> >      // was built with.
> >      size_t loaded_version = dynload::cudnnGetVersion();
> > -    bool library_loaded_matches_source = (loaded_version == CUDNN_VERSION);
> > +    size_t loaded_major_version = cudnnMajorVersion(loaded_version);
> > +    bool library_loaded_matches_source =
> > +        (loaded_major_version == cudnnMajorVersion(CUDNN_VERSION));
>
> So if I divide by CUDNN_VERSION by 100 instead of 1000 and rename to
> cudnnCompatibilityVersion, would that be sufficient?
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/tensorflow/tensorflow/pull/3249/files/1416889464a7e2ea1ddfb773c07fc5f9f0c7cac8#r70566758>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/APAgTri4SHyzrittdhbrMi0dVH4VixCUks5qVHr6gaJpZM4JIuJc>
> .
>